protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java

```diff
@@ -0,0 +1,471 @@
+package org.geysermc.mcprotocollib.auth;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import org.geysermc.mcprotocollib.auth.util.UndashedUUIDAdapter;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyFactory;
+import java.security.PublicKey;
+import java.security.Signature;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * Information about a user profile.
+ */
+public class GameProfile {
+    private static final String[] WHITELISTED_DOMAINS = {".minecraft.net", ".mojang.com"};
+    private static final PublicKey SIGNATURE_KEY;
+    private static final Gson GSON;
+
+    static {
+        try (InputStream in = Objects.requireNonNull(SessionService.class.getResourceAsStream("/yggdrasil_session_pubkey.der"))) {
+            SIGNATURE_KEY = KeyFactory.getInstance("RSA")
+                    .generatePublic(new X509EncodedKeySpec(in.readAllBytes()));
+        } catch (Exception e) {
+            throw new ExceptionInInitializerError("Missing/invalid yggdrasil public key.");
+        }
+
+        GSON = new GsonBuilder().registerTypeAdapter(UUID.class, new UndashedUUIDAdapter()).create();
+    }
+
+    private static boolean isWhitelistedDomain(String url) {
+        URI uri;
+        try {
+            uri = new URI(url);
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException("Invalid URL \"" + url + "\".");
+        }
+
+        String domain = uri.getHost();
+        for (String whitelistedDomain : WHITELISTED_DOMAINS) {
+            if (domain.endsWith(whitelistedDomain)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private final UUID id;
+    private final String name;
+
+    private List<Property> properties;
+    private Map<TextureType, Texture> textures;
+    private boolean texturesVerified;
```

Comment:
These should be above the static init block

Reply:
No, other classes like ServerListener have the non-static fields below the static block.

Reply:
Ah ok, something to likely look at in future then